physical/raft/raft_autopilot.go

```diff
@@ -428,6 +429,17 @@ func (d *Delegate) KnownServers() map[raft.ServerID]*autopilot.Server {
 			Ext:         d.autopilotServerExt(state),
 		}
 
+		// As KnownServers is a delegate called by autopilot let's check if we already had this data in the correct format and use it.
+		// If we don't (which sounds a bit sad, unless this ISN'T a voter) then as a fail-safe, let's try what we've done elsewhere
+		// in code to check the desired suffrage and manually set NodeType based on whether that's a voter or not.
+		// If we don't  do either of these things, NodeType isn't set which means technically it's not a voter.
+		// It shouldn't be a voter and end up in this state.
+		if apServerState, found := apServerStates[raft.ServerID(id)]; found && apServerState.Server.NodeType != "" {
+			server.NodeType = apServerState.Server.NodeType
+		} else if state.DesiredSuffrage != "non-voter" {
```

Comment:
The two reasons I can think of are:

1. The code I came across which checked `state.DesiredSuffrage` compared against `non-voter` so I just figured, keep it similar.
2. There was a part of me that wasn't 100% convinced there might ever be a case where it could have been something else that was a 'voter' but had some other name.

Looking into 2 again just now I guess I could just flip the check to `else if state.DesiredSuffrage == "voter" {`
https://github.com/hashicorp/vault/blob/main/vault/logical_system_raft.go#L364-L370